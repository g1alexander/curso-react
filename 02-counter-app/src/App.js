import PropTypes from "prop-types";

export function App({ saludo, subtitulo }) {
  return (
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo}</p>
    </>
  );
}

App.propTypes = {
  saludo: PropTypes.string.isRequired,
};

App.defaultProps = {
  subtitulo: "Soy subtitulo",
};
