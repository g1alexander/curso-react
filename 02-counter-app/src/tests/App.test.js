// import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import { App } from "../App";

describe("test component <App />", () => {
  // test('debe mostrar el mensaje "Hola, Soy Goku"', () => {
  //   const saludo = "Hola, Soy Goku";
  //   const { getByText } = render(<App saludo={saludo} />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });

  test("debe mostrar <App /> correctamente", () => {
    const saludo = "Hola, Soy Goku";
    const wrapper = shallow(<App saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el subtitulo mandado por props", () => {
    const saludo = "Hola, Soy Goku";
    const subTitulo = "Soy subtitulo";
    const wrapper = shallow(<App saludo={saludo} subtitulo={subTitulo} />);

    const $p = wrapper.find("p").text();

    //wrapper.find() funciona igual a document.querySelector()

    expect($p).toBe(subTitulo);
  });
});
