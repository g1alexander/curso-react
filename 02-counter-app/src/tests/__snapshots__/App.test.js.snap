// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component <App /> debe mostrar <App /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    Soy subtitulo
  </p>
</Fragment>
`;
